### PR TITLE
cleanup l_net

### PR DESCRIPTION
**Pull request by Vorschreibung**
Various cleanups for `l_net`.

This also fixes a standing issue that these UDP sockets were never
closed and each build invocation accumulated lingering unconnected UDP
sockets.

### CODE REVIEW

**Comment by Vorschreibung** [comment]
Ready to review :blush:, to verify:

```
grep WINS_MyAddress . -r
grep Net_MyAddress . -r
```

Tested on Linux.

File: libs/l_net/l_net.c [hunk 1]
```diff
@@ -307,19 +307,8 @@ void Net_StringToAddress( const char *string, address_t *address ){
...
-//===========================================================================
-//
-// Parameter:				-
-// Returns:					-
-// Changes Globals:		-
-//===========================================================================
 void Net_Setup( void ){
 	WINS_Init();
-	//
-	WinPrint( "my address is %s\n", WINS_MyAddress() );
```
Comment (Vorschreibung) [inline libs/l_net/l_net.c:322]: The only usage of the code to be removed.

File: libs/l_net/l_net_berkley.c [hunk 2]
```diff
@@ -193,43 +190,18 @@ int WINS_Init( void ){
 //		Cvar_Set ("hostname", buff);
 	}
 
-	//++timo WTF is that net_controlsocket? it's sole purpose is to retrieve the local IP?
```
Comment (Vorschreibung) [inline libs/l_net/l_net_berkley.c:196]: timo pointed towards the right direction :blush:

**Comment by Vorschreibung** [comment]
Ready to review :blush:, to verify:

```
grep WINS_MyAddress . -r
grep Net_MyAddress . -r
```

Tested on Linux.

**Comment by Garux** [comment]
Added some tweaks, want to know your opinion.

**Comment by Vorschreibung** [comment]
> Added some tweaks, want to know your opinion.

Good catch! I've dug deeper and it turns out that `broadcastaddr` as well was not needed, turns out the whole broadcasting over UDP was just used to get one's own IP address.

I've followed this trail and continued removing even more functions that were all unused now: https://github.com/Garux/netradiant-custom/pull/204/commits/cb54f917786c74606706881197609f25f442e908#diff-be9a2b010ddc8f46a80d007c0c991216faf0a3501574d938afb1f65e1d9b34d3

---

I've also noticed that the 2 `l_net_wins` implementations (windows and posix) are mostly identical now and thus have merged them into one with ifdefs: https://github.com/Garux/netradiant-custom/pull/204/commits/7df46978c8556d89ce27ba47e4b38b62d90d440b

Please let me know if you think this is a good approach.

@Garux while I was careful with the changes, I have verified and tested only under Linux, if you can test under Windows I would appreciate, otherwise please tell me and I'll test on Windows myself

File: libs/l_net/l_net_wins.c [hunk 3]
```diff
@@ -35,409 +67,409 @@ typedef struct tag_error_struct
...
 #define MAXHOSTNAMELEN      256
 
 static unsigned long myAddr;
 
+#ifdef WIN32
 WSADATA winsockdata;
+#endif
 
 ERROR_STRUCT errlist[] = {
-	{WSAEINTR,           "WSAEINTR - Interrupted"},
```
Comment (Vorschreibung) [inline libs/l_net/l_net_wins.c:66]: note: these diffs are due to applying `uncrustify`

File: libs/l_net/l_net_wins.c [hunk 4]
```diff
@@ -35,409 +67,409 @@ typedef struct tag_error_struct
...
-// Changes Globals:		-
-//===========================================================================
-int WINS_OpenReliableSocket( int port ){
+}
+
+int WINS_OpenReliableSocket(int port)
+{
 	int newsocket;
 	struct sockaddr_in address;
-	BOOL _true = 0xFFFFFFFF;
```
Comment (Vorschreibung) [inline libs/l_net/l_net_wins.c:326]: note: I've checked, `BOOL` is equal to `int` on Windows

**Comment by Vorschreibung** [comment]
> Added some tweaks, want to know your opinion.

Good catch! I've dug deeper and it turns out that `broadcastaddr` as well was not needed, turns out the whole broadcasting over UDP was just used to get one's own IP address.

I've followed this trail and continued removing even more functions that were all unused now: https://github.com/Garux/netradiant-custom/pull/204/commits/cb54f917786c74606706881197609f25f442e908#diff-be9a2b010ddc8f46a80d007c0c991216faf0a3501574d938afb1f65e1d9b34d3

---

I've also noticed that the 2 `l_net_wins` implementations (windows and posix) are mostly identical now and thus have merged them into one with ifdefs: https://github.com/Garux/netradiant-custom/pull/204/commits/7fa6c48bf556b173e3d422e9182db2b7f5d8cd3d

Please let me know if you think this is a good approach.

@Garux while I was careful with the changes, I have verified and tested only under Linux, if you can test under Windows I would appreciate, otherwise please tell me and I'll test on Windows myself

**Comment by Garux** [comment]
> @Garux while I was careful with the changes, I have verified and tested only under Linux, if you can test under Windows I would appreciate, otherwise please tell me and I'll test on Windows myself

Sorry for the delay, monitoring works in Windows here, in successful case at least.

**Comment by Vorschreibung** [comment]
> > @Garux while I was careful with the changes, I have verified and tested only under Linux, if you can test under Windows I would appreciate, otherwise please tell me and I'll test on Windows myself
> 
> Sorry for the delay, monitoring works in Windows here, in successful case at least.

No problem, I've also tested on Windows and I think this is looking good and can be merged :metal:

**Comment by Garux** [comment]
> No problem, I've also tested on Windows and I think this is looking good and can be merged 🤘

Yeah, functionality seems fine. Two problems are pending wrt uncrustify commit.

**Comment by Vorschreibung** [comment]
@Garux sorry for the delay from my side and the failed `uncrustify` - I applied wrongly previously, now it should be correct :metal: :blush:

**Comment by Garux** [comment]
Fixed a couple of inaccuracies in uncrustify commit manually.

**Pull request merged**